app/backend/approaches/chatreadretrieveread.py

```diff
@@ -133,19 +137,52 @@ async def run_until_final_call(
             new_user_content=user_query_request,
             max_tokens=self.chatgpt_token_limit - query_response_token_limit,
         )
+        
+        if self.use_hugging_face:
+            huggingf_search_prompt = ""
+            
+            for message in query_messages:
+                huggingf_search_prompt += message["role"] + ": "
+                huggingf_search_prompt += message["content"] + '\n'
+                                
+            API_URL = os.gentenv("HUGGINGFACE_API_URL")
+            API_KEY = os.getenv("HUGGINGFACE_API_KEY")
+            headers = {"Authorization": "Bearer " + API_KEY}
+            
+            def ask_huggingface(payload):
+                response = requests.post(API_URL, headers=headers, json=payload)
+                return response.json()
```

Comment:
Looks like these things could be refactored:
* We're reusing them in `retrievethenread.py`; no need to repeat them there
* This is not something related to these specific classes. Perhaps we could introduce a module for Hugging Face utils and abstract all of the HF related functionality there.